src/frontend/Parser.y

```diff
@@ -1057,6 +1058,49 @@ checkIndexAccess array index tk = do
       RWS.tell [Error "Left side is not a chest" (T.position tk)]
       return T.TypeError
 
+proceduresCheck :: G.Id -> [G.Expr] -> T.Token -> ST.ParserMonad ([G.Expr])
+proceduresCheck procId exprs tk = do
+  procType <- getType procId
+  case procType of
+    T.ProcedureT domain -> do
+      let exprsTypes = exprsToTypes exprs
+      let exprsTypesL = length exprsTypes
+      let domainL = length domain
+      if exprsTypesL < domainL then do
+        RWS.tell [Error ("Procedure " ++ show procId ++ " received less arguments than it expected") (T.position tk)]
+        return exprs
+      else if exprsTypesL > domainL then do
+        RWS.tell [Error ("Procedure " ++ show procId ++ " received more arguments than it expected") (T.position tk)]
+        return exprs
+      else do
+        if T.TypeError `elem` exprsTypes then
+          return exprs
+        else case findInvalidArgument $ zip exprsTypes domain of
+          -- all looking good, now cast each argument to it's correspondent parameter type
+          Nothing -> do
+            let castedExprs = addCastToExprs $ zip exprs domain
+            return (castedExprs)
+
+          -- oh oh, an argument could not be implicitly casted to its correspondent argument
+          Just x -> do
+            RWS.tell [Error ("Argument #" ++ show x ++ " type mismatch with parameter #" ++ show x ++ " type") (T.position tk)]
+            return exprs
+
+    T.TypeError -> return exprs
+    _ -> do
+      RWS.tell [Error "You're trying to call a non-callable expression" (T.position tk)]
+      return exprs
+  where
+    -- left side is the type of the argument, right side is the parameter type
+    findInvalidArgument :: [(T.Type, T.Type)] -> Maybe Int
+    findInvalidArgument ts = findInvalid ts 0
```

Comment:
This function is the same as `functionsCheck`. Can't be reuse that function here? They are doing literally the same